### PR TITLE
Merge with reflector

### PR DESCRIPTION
**Pull request by lyoshenka**
this is still missing the command that starts reflector itself

### CODE REVIEW

File: player/hot_cache.go [hunk 1]
```diff
@@ -6,41 +6,34 @@ import (
...
+		return nil
 	}
-	return nil
+	return cachedItem.Value().(ReadableChunk)
 }
 
-func (h *HotCache) Set(key string, chunk *reflectedChunk) {
-	h.cache.Set(key, *chunk, h.ttl)
+func (h *HotCache) Set(hash string, chunk ReadableChunk) {
+	h.cache.Set(hash, chunk, longTTL)
```
Comment (nikooo777) [inline player/hot_cache.go:28]: I think there was a valid reason as to why I didn't pass it as pointer. I can't recall why but can you think of anything?

**Review by nikooo777 — COMMENTED** [review]
i see you moved around a lot of code. did you change any of it? because the diff doesn't show if you did.
I reviewed the changes only

File: go.mod [hunk 2]
```diff
@@ -2,6 +2,8 @@ module github.com/lbryio/lbrytv-player
 
 go 1.15
 
+replace github.com/btcsuite/btcd => github.com/lbryio/lbrycrd.go v0.0.0-20200203050410-e1076f12bf19
```
[the remark this answers is not in the document]
Comment (lyoshenka) [reply go.mod:5]: it didn't work for me without this

File: player/hot_cache.go (hunk 1 above)
Comment (lyoshenka) [reply player/hot_cache.go:28]: ReadableChunk used to be an interface, with *reflectedChunk implementing that interface.

Now that ReadableChunk is a byte-slice, it's passed by reference anyway.

**Comment by lyoshenka** [comment]
@andybeletsky i rearranged a lot of stuff, if you're curious

File: cmd/main.go [hunk 3]
```diff
@@ -4,118 +4,195 @@ import (
...
+}
+
+func diskCacheParams() (int, string) {
+	l := Logger
+
+	if diskCacheConfig == "" {
+		return 0, ""
+	}
+
+	parts := strings.Split(diskCacheConfig, ":")
```
[the remark this answers is not in the document]
Comment (lyoshenka) [reply cmd/main.go:145]: what do you recommend for the case where a feature (e.g. disk cache) is disabled and if you want to enable it, you have to pass multiple flags (e.g. cache size and path to store the blobs). what's the right way to document that so it's clear that you must have both or neither

File: cmd/main.go [hunk 4]
```diff
@@ -4,118 +4,195 @@ import (
...
+	var hotCacheBytes datasize.ByteSize
+	err := hotCacheBytes.UnmarshalText([]byte(hotCacheSize))
+	if err != nil {
+		Logger.Fatal(err)
+	}
+	if hotCacheBytes <= 0 {
+		Logger.Fatal("hot cache size must be greater than 0. if you want to disable hot cache, you'll have to do a bit of coding")
+	}
+
+	avgSDBlobSize := 1000      // JUST A GUESS (in bytes)
```
[the remark this answers is not in the document]
Comment (lyoshenka) [reply cmd/main.go:99]: originally i wanted to run some analysis on existing sd blobs

but now i think i have a better way to do this so we don't have to guess.

File: cmd/main.go [hunk 5]
```diff
@@ -4,118 +4,195 @@ import (
...
+	metrics.InstallRoute(a.Router)
+	if enableProfile {
+		player.InstallProfilingRoutes(a.Router)
+	}
+
+	a.Start()
+	a.ServeUntilShutdown()
+}
+
+func initHotCache(origin store.BlobStore) *player.HotCache {
```
[the remark this answers is not in the document]
Comment (lyoshenka) [reply cmd/main.go:89]: where would you put this?

File: cmd/main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (andybeletsky) [reply cmd/main.go:145]: What if we simply set some default small cache size (like 500MB) that kicks in if cache is enabled? It’s going to be explicitly set in production anyway, and for development purposes there’s no need think about that.

File: cmd/main.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (andybeletsky) [reply cmd/main.go:89]: inside player.NewHotCache perhaps

File: cmd/main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (lyoshenka) [reply cmd/main.go]: done, ty

**Comment by andybeletsky** [comment]
As a suggestion, I think it would increase the review-ability of pull requests greatly if large code reorgs (like deleting/creating large source code files or merging external packages) are kept separately from small refactoring

**Comment by lyoshenka** [comment]
> As a suggestion, I think it would increase the review-ability of pull requests greatly if large code reorgs (like deleting/creating large source code files or merging external packages) are done separately from small refactoring.

I did put them in separate commits, but you're right that its still hard to read on github. I'll try to rebase those down the commit stack and merge them into master on their own. Sometimes its hard though.

**Comment by lyoshenka** [comment]
merged in 4197ca4cfc828de088ff3f43d5c918eff76bd99b

**Pull request closed without merging**